### PR TITLE
Migration fixes

### PR DESCRIPTION
**Pull request by MechaRyRy**
### Problem
The migration demo is a little broken. It did not take into account:
- current and total bytes do not match, resolving in a partial. 
- one file matches another from v1, forcing an empty file for the second usage as the first is deleted
- We didn't move the completed batch files

### Solution
Update the database and fix the above issues.

### CODE REVIEW

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/CompletedDownloadBatchesExtractor.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
             + "batches INNER JOIN DownloadsByBatch ON DownloadsByBatch.batch_id = batches._id "
             + "WHERE batches._id NOT IN (SELECT DownloadsByBatch.batch_id FROM DownloadsByBatch "
             + "INNER JOIN batches ON batches._id = DownloadsByBatch.batch_id "
-            + "WHERE DownloadsByBatch._data IS NULL "
+            + "WHERE DownloadsByBatch.batch_total_bytes != DownloadsByBatch.batch_current_bytes "
+            + "OR DownloadsByBatch._data IS NULL "
```
Comment (danybony) [inline demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/CompletedDownloadBatchesExtractor.java:31]: 💯

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java [hunk 2]
```diff
@@ -66,24 +72,72 @@ public void run() {
...
+            CompletedDownloadBatch completedDownloadBatch = completeDownloadBatches.get(i);
+            downloadManager.addCompletedBatch(completedDownloadBatch);
+            completeDownloadBatches.remove(i);
+
+            for (CompletedDownloadFile completedDownloadFile : completedDownloadBatch.completedDownloadFiles()) {
+                migrateV1FileToV2Location(completedDownloadFile);
+
+                boolean usedInFurtherBatches = originalFileUsedInFurtherBatches(
+                        completedDownloadFile.originalFileLocation(),
+                        partialDownloadBatches,
```
Comment (danybony) [inline demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java:104]: At this point `partialDownloadBatches` is empty, right?

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java [hunk 3]
```diff
@@ -66,24 +72,72 @@ public void run() {
...
         database.deleteDatabase();
         onUpdate("Completed Migration");
     }
 
+    private boolean originalFileUsedInFurtherBatches(String originalFile,
+                                                     List<VersionOnePartialDownloadBatch> partialDownloadBatches,
+                                                     List<CompletedDownloadBatch> completeDownloadBatches) {
+
+        for (VersionOnePartialDownloadBatch partialDownloadBatch : partialDownloadBatches) {
+            for (String originalFileLocation : partialDownloadBatch.originalFileLocations()) {
```
Comment (danybony) [inline demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java]: here you can simply do 
```
 if (partialDownloadBatch.originalFileLocations().contains(originalFile)) {
       return true;
}
```

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java (hunk 2 above)
Comment (MechaRyRy) [reply demo-simple/src/main/java/com/novoda/downloadmanager/demo/migration/MigrationJob.java:104]: Yea should be, I didn't want to create another method that did the same thing but had one list less 😂

**Review by danybony — APPROVED** [review]

**Pull request merged**